src/modules/httptrap.c

```diff
@@ -533,7 +537,7 @@ httptrap_yajl_cb_end_map(void *ctx) {
        * in progress metrics only (get_metric) not (get_last_metric)
        * and we also much fetch a count...
        */
-      track_filtered(json, metric_name);
+      track_filtered(json, &metric_name);
```

Comment:
who frees ``metric_name`` if the value was changed?

Reply:
`track_filtered` calls `noit_apply_filterset`, which will free it.